lib/charms/ovn_charm.py

```diff
@@ -188,6 +188,34 @@ def __init__(self, **kwargs):
                 self.restart_map.update({
                     '/etc/tmpfiles.d/nova-ovs-vhost-user.conf': []})
 
+    def is_port(self, port, status_set=True):
+        """Check if the port exists.
+
+        :param port: interface name
+        :type port: str
+        :param status_set: change the juju state to blocked if
+                           the port does not exist
+        :type status_set: bool
+        :returns: boolean whether port exists
+        :rtype: bool
+        :raises: ValueError
+        """
+        ch_core.hookenv.log('checking port "{}" existence'.format(port),
+                            level=ch_core.hookenv.DEBUG)
+        try:
+            if self.options.enable_dpdk:
+                # TODO: add function to check DPDK interface
+                pass
+            else:
+                self.run('ip', 'link', 'show', port)
+        except subprocess.CalledProcessError:
+            message = 'port "{}" does not exists'.format(port)
+            ch_core.hookenv.log(message, level=ch_core.hookenv.ERROR)
+            if status_set:
+                ch_core.hookenv.status_set('blocked', message)
```

Comment:
While this technically does set the status for the charm the framework the OpenStack charms is written in will also attempt to update status at the end of every hook execution. Thus I would use that framework to set the status rather than using the hook tool directly.

You have a couple of options:

This could be accomplished by either setting a reactive flag or a class instance variable and pick this up here:
https://github.com/openstack-charmers/charm-layer-ovn/blob/e987e12ec92a00d630ddc0eaf5823001b7429910/lib/charms/ovn_charm.py#L240-L274

Another option is to override and use the `custom_assess_status_last_check` method:
https://github.com/openstack/charms.openstack/blob/8b2eed06c4070cc7d242d11e51747c14a283f43c/charms_openstack/charm/core.py#L1217

Reply:
Thank you, this looks like a much better solution.